PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/sqlserver/CreateSqlServerAction.java

```diff
@@ -0,0 +1,93 @@
+import com.microsoft.azure.toolkit.lib.sqlserver.SqlServerConfig;
+import com.microsoft.azure.toolkit.lib.sqlserver.SqlServerService;
+import com.microsoft.azure.toolkit.lib.sqlserver.service.ISqlServer;
+import com.microsoft.azuretools.authmanage.AuthMethodManager;
+import com.microsoft.azuretools.utils.AzureUIRefreshCore;
+import com.microsoft.azuretools.utils.AzureUIRefreshEvent;
+import com.microsoft.intellij.AzurePlugin;
+import com.microsoft.intellij.actions.AzureSignInAction;
+import com.microsoft.intellij.util.AzureLoginHelper;
+import com.microsoft.tooling.msservices.components.DefaultLoader;
+import com.microsoft.tooling.msservices.helpers.Name;
+import com.microsoft.tooling.msservices.serviceexplorer.AzureActionEnum;
+import com.microsoft.tooling.msservices.serviceexplorer.Node;
+import com.microsoft.tooling.msservices.serviceexplorer.NodeActionEvent;
+import com.microsoft.tooling.msservices.serviceexplorer.NodeActionListener;
+import com.microsoft.tooling.msservices.serviceexplorer.azure.sqlserver.SqlServerModule;
+
+import static com.microsoft.intellij.ui.messages.AzureBundle.message;
+
+@Name("Create")
+public class CreateSqlServerAction extends NodeActionListener {
+
+    private final SqlServerModule model;
+
+    public CreateSqlServerAction(SqlServerModule model) {
+        super();
+        this.model = model;
+    }
+
+    @Override
+    public AzureActionEnum getAction() {
+        return AzureActionEnum.CREATE;
+    }
+
+    @Override
+    public void actionPerformed(NodeActionEvent e) {
+        final Project project = (Project) model.getProject();
+        AzureSignInAction.doSignIn(AuthMethodManager.getInstance(), project).subscribe((isSuccess) -> {
+            this.doActionPerformed(e, isSuccess, project);
+        });
+    }
+
+    private void doActionPerformed(NodeActionEvent e, boolean isLoggedIn, Project project) {
+        try {
+            if (!isLoggedIn ||
+                !AzureLoginHelper.isAzureSubsAvailableOrReportError(message("common.error.signIn"))) {
+                return;
+            }
+        } catch (final Exception ex) {
+            AzurePlugin.log(message("common.error.signIn"), ex);
+            DefaultLoader.getUIHelper().showException(message("common.error.signIn"), ex, message("common.error.signIn"), false, true);
+        }
+        final SqlServerCreationDialog dialog = new SqlServerCreationDialog(project);
+        dialog.setOkActionListener((data) -> this.createSqlServer(data, dialog));
+        dialog.show();
+    }
+
+    private void createSqlServer(final SqlServerConfig config, SqlServerCreationDialog dialog) {
+        final Runnable runnable = () -> {
+            final ProgressIndicator indicator = ProgressManager.getInstance().getProgressIndicator();
+            indicator.setIndeterminate(true);
+            DefaultLoader.getIdeHelper().invokeLater(dialog::close);
+            ISqlServer server = SqlServerService.getInstance().create(config);
```

Comment:
SqlServerService is a bit confusing

Reply:
the class is not in this PR. Will rename it to CreateSqlServerTask in next PR.

Reply:
keep this comment unsolved until new PR is merged

Reply:
updated in https://github.com/microsoft/azure-tools-for-java/commit/ce6109b33be79c9d8552a3d1fedd83c86531e7bb